src/preview/topics.rs

```diff
@@ -0,0 +1,292 @@
+use momento_protos::cache_client::pubsub::{
+    self, pubsub_client::PubsubClient, PublishRequest, SubscriptionRequest,
+};
+use tonic::{codegen::InterceptedService, transport::Channel};
+
+use crate::MomentoError;
+use crate::{
+    endpoint_resolver::MomentoEndpointsResolver,
+    grpc::header_interceptor::HeaderInterceptor,
+    utils::{connect_channel_lazily, user_agent},
+    MomentoResult,
+};
+
+type ChannelType = InterceptedService<Channel, HeaderInterceptor>;
+
+pub struct Pubsub {
+    client: pubsub::pubsub_client::PubsubClient<ChannelType>,
+}
+
+/// Work with topics, publishing and subscribing.
+impl Pubsub {
+    pub fn connect(
+        auth_token: String,
+        momento_endpoint: Option<String>,
+        user_application_name: Option<&str>,
+    ) -> MomentoResult<Self> {
```

Comment:
If you do like having connect as an explicit API, and it's just returning Self, then would it make more sense to have it be a static factory method instead of requiring a constructor call followed by a connect call?

Reply:
oh, looking at your example code it kind of looks like that's already the pattern you are intending and I am just still very bad at reading rust code :(

Reply:
It's cool, I see lots of clients using static factory methods named like "connect" instead of a sterile "new" and I kind of like that. But the struct name I chose is not helpful in revealing that intent so yeah TopicClient::connect() for the win